dependency/resolver.go

```diff
@@ -283,15 +284,37 @@ func (r *Resolver) ResolveAll(deps []*cfg.Dependency) ([]string, error) {
 // resolveList takes a list and resolves it.
 func (r *Resolver) resolveList(queue *list.List) ([]string, error) {
 
+	res := make([]string, 0, queue.Len())
+
 	var failedDep string
 	for e := queue.Front(); e != nil; e = e.Next() {
 		dep := e.Value.(string)
-		t := strings.TrimPrefix(e.Value.(string), r.VendorDir+string(os.PathSeparator))
-		if r.Config.HasIgnore(t) {
-			msg.Info("Ignoring: %s", t)
+		importPath := r.vendorToImportPath(dep)
+
+		if r.SpecificSubpackages {
+			// If subpackages are specified, process only them, not the root
+			existing := r.Config.Imports.Get(importPath)
+			if existing != nil && len(existing.Subpackages) > 0 {
+				// Build the list of subpackages
+				sublist := list.New()
+				for _, subPkg := range existing.Subpackages {
+					sublist.PushBack(filepath.Join(dep, filepath.FromSlash(subPkg)))
+				}
+				// Process it
+				subRes, err := r.resolveList(sublist)
+				if err != nil {
+					return []string{}, err
+				}
+				res = append(res, subRes...)
+				continue
+			}
+		}
```

Comment:
This is looking at the sub-packages in the config while still walking the directory tree (`filepath.Walk` below).

Instead of walking the file tree for packages in the `vendor/` folder what about walking the Import tree? This is what @technosophos is working on at the moment.

I'd like to hold off on this until we get to see what that looks like so we can compare them.